pandas/tests/test_groupby.py

```diff
@@ -6104,6 +6104,20 @@ def test_nunique_with_object(self):
         expected = pd.Series([1] * 5, name='name', index=index)
         tm.assert_series_equal(result, expected)
 
+    def test_transform_with_non_scala_group(self):
+        # GH 10165
+        cols = pd.MultiIndex.from_tuples([
+            ('syn', 'A'), ('mis', 'A'), ('non', 'A'),
+            ('syn', 'C'), ('mis', 'C'), ('non', 'C'),
+            ('syn', 'T'), ('mis', 'T'), ('non', 'T'),
+            ('syn', 'G'), ('mis', 'G'), ('non', 'G')])
+        df = pd.DataFrame(np.random.randint(1, 10, (4, 12)),
+                          columns=cols,
+                          index=['A', 'C', 'G', 'T'])
+        self.assertRaises(ValueError,
```

Comment:
use ``self.assertRaisesRegex`` to match the generated error message (or part of it)